metadata_backend/database/db_service.py

```diff
@@ -155,3 +155,15 @@ async def get_count(self, collection: str, query: Dict) -> int:
         """
         LOG.debug("DB doc count performed.")
         return await self.database[collection].count_documents(query)
+
+    @auto_reconnect
+    async def create_folder(self, folder_data: Dict) -> Tuple[bool, str]:
```

Comment:
This is almost the same code than in `create` method below, maybe just modify it a bit, e.g.
```
async def create(self, collection: str, document: Dict, return_folder_id: bool = False) -> Union[bool, string]:
...
if return_folder_id:
    return return.inserted_id
```

Also, you don't need to check if result was acknowledged. If id is returned, operation was successful, if None / error was raised, it wasn't. 

Acknowledgion is checked in metadata object methods, because we don't use the mongodb internal id in our responses (however, I'm not even sure if that acknowledgion check is the best way to do that...).